sympy/polys/polytools.py

```diff
@@ -3870,6 +3870,30 @@ def cancel(f, g, include=False):
         else:
             return tuple(map(per, result))
 
+    def gal(f, by_name=False, max_tries=30, randomize=False):
```

Comment:
I think that the name could be 'galois'. That is not too long.

Reply:
Sure. I was only thinking of `gal()` because of the notation $Gal(E/F)$. But I suppose `galois()` might be more easily discoverable by new users.

Reply:
Also a reasonable perspective. I'm honestly fine with any of these options. I guess tab-completion in IDEs means shortness is really a non-issue.

Reply:
There are many precedents for `Poly` methods having the exact same name as sympy functions that operate on Expr:
```
In [1]: d = {}

In [2]: exec('from sympy import *', d)

In [3]: len(d)
Out[3]: 913

In [4]: d.keys() & set(dir(Poly))
Out[4]: 
{'LC',
 'LM',
 'LT',
 'cancel',
 'cofactors',
 'compose',
 'content',
 'count_ops',
 'count_roots',
 'decompose',
 'degree',
 'degree_list',
 'diff',
 'discriminant',
 'div',
 'exquo',
 'factor_list',
 'gcd',
 'gcdex',
 'gff_list',
 'ground_roots',
 'half_gcdex',
 'homogeneous_order',
 'integrate',
 'intervals',
 'invert',
 'lcm',
 'monic',
 'nroots',
 'nth_power_roots_poly',
 'pdiv',
 'per',
 'pexquo',
 'pquo',
 'prem',
 'primitive',
 'quo',
 'real_roots',
 'refine',
 'refine_root',
 'rem',
 'resultant',
 'root',
 'simplify',
 'sqf_list',
 'sqf_norm',
 'sqf_part',
 'sturm',
 'subresultants',
 'terms_gcd',
 'total_degree',
 'trunc'}
```
Not all of those are equivalent but most are.

Reply:
Interesting. Maybe the `galois_group` function should be made to fit into the existing pattern as well.

Reply:
I've adapted it to fit the general pattern. The main implementation is now in the `Poly` method. The top-level function accepts `Expr`, converts to `Poly`, and uses the `Poly` method.

Reply:
Also the method names in `Poly` and `AlgebraicField` are now `galois_group`, matching the top-level function.